extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -702,20 +699,48 @@ public String getErrorMessage() {
     @Override
     public void onFetchPage(@Nonnull final Downloader downloader)
             throws IOException, ExtractionException {
-        initialAjaxJson = getJsonResponse(getUrl() + "&pbj=1", getExtractorLocalization());
 
-        initialData = initialAjaxJson.getObject(3).getObject("response", null);
-        if (initialData == null) {
-            initialData = initialAjaxJson.getObject(2).getObject("response", null);
+        final String url = "https://youtubei.googleapis.com/youtubei/v1/player?key=" + getKey();
+
+        // @formatter:off
+        byte[] json = JsonWriter.string()
+                .object()
+                .object("context")
+                    .object("client")
+                        .value("clientName", "Android")
+                        .value("clientVersion", "16.02")
```

Comment:
Can you try updating the client version to the latest release version (16.08 from APKMirror) and see if URLs are not encrypted?

Reply:
16.02 is the version that YouTube Vanced uses, but yes, 16.06 does work as well.